modules/connectionquality/connectionquality.js

```diff
@@ -35,22 +36,78 @@ function calculateQuality(newVal, oldVal) {
     return (newVal <= oldVal) ? newVal : (9*oldVal + newVal) / 10;
 }
 
+// webrtc table describing simulcast resolutions and used bandwidth
+// https://chromium.googlesource.com/external/webrtc/+/master/webrtc/media/engine/simulcast.cc#42
+const _bandwidthMap = [
+    { width: 1920, height: 1080, layers:3, max: 5000, min: 800 },
+    { width: 1280, height: 720,  layers:3, max: 2500, min: 600 },
+    { width: 960,  height: 540,  layers:3, max: 900,  min: 450 },
+    { width: 640,  height: 360,  layers:2, max: 700,  min: 150 },
+    { width: 480,  height: 270,  layers:2, max: 450,  min: 150 },
+    { width: 320,  height: 180,  layers:1, max: 200,  min: 30 }
+];
+
+/**
+ * We disable quality calculations based on bandwidth if simulcast is disabled,
+ * or enable it in case of no simulcast and we force it.
+ * @type {boolean}
+ */
+const disableQualityBasedOnBandwidth =
+    config.forceQualityBasedOnBandwidth ? false : config.disableSimulcast;
+
+/**
+ * Calculates the quality percentage based on the input resolution height and
+ * the upload reported by the client. The value is based on the interval from
+ * _bandwidthMap.
+ * @param inputHeight the resolution used to open the camera.
+ * @param upload the upload rate reported by client.
+ * @returns {*} the percent of upload based on _bandwidthMap and maximum value
```

Comment:
You can set the returned type to int and add some explanation about undefined results.